src/App.jsx

```diff
@@ -1,26 +1,195 @@
 import { useState } from 'react'
+import { useEffect } from 'react'
 import './App.css'
 import './stylesheets/movielist.css'
 
 import MovieList from './components/MovieList'
 import Header from './components/Header'
 import Footer from './components/Footer'
 
+const baseDetailsUrl = 'https://api.themoviedb.org/3/movie/now_playing?language=en-US'; //url that we are sending fetch request to that returns array with movies now playing and details without page at end
+const baseSearchUrl = 'https://api.themoviedb.org/3/search/movie'; //url that is the request that returns movie results depending on a query parameter which we are getting from our SearchBar component
 
 const App = () => {
+
+  const [movieList, setMovieList] = useState([]); //decalre a movieList as an empty array. We will populate it with a fetch request to imdb API
+  const [pageNum, setPageNum] = useState(1); //pageNum will keep track of the movies we are rendering. Whenever we click the load more button the page increases and we request next page of movies
+  const [isSearching, setIsSearching] = useState(false) //flag to know whether or not we are searching to call correct function in useEffect
+  const [searchTerm, setSearchTerm] = useState(''); //need this so that searching pagination also works
+
+
+  //function just updates the state of the pageNum everytime user clicks load more button.
+  function handleLoadMoreClick(){
+    setPageNum((pageNum) => pageNum + 1);
+  }
+
+  //function makes fetch request to movie API depending on what page we need to load. If we are loading more pages past 1 we append what fetch request retrusn to our existing array
+  async function getMovieList(pageIdx){
+    
+    const urlWithPage = `${baseDetailsUrl}&page=${pageIdx}` //update url depending on the pageIdx that gets passed in which is being updated depending on userClick
+  
+
+    try{
+      const res = await fetch(urlWithPage, {
+        headers:{
+          accept: 'application/json',
+          Authorization: `bearer ${import.meta.env.VITE_API_KEY}`,
+    
+        },
+      });
+      if(!res.ok){
+        throw new Error('Bad api request');
+      }
+
+      const movieList = await res.json();
+      const newMovies = movieList.results; //gives us the actual movie array
+
+      setMovieList( (prevList) => [...prevList, ...newMovies]); //spread operator ... allows us to join contents of two different arrays together. I.E a = [1,2,3] b = [4,5] c = [...a,...b] c = [1,2,3,4,5]
```

Comment:
You have a function that does it for you and is also faster 👀 
```suggestion
      setMovieList( (prevList) => prevList.concat(newMovies); 
```